src/server/game/Entities/Unit/Unit.h

```diff
@@ -2297,6 +2305,9 @@ class TC_GAME_API Unit : public WorldObject
         typedef std::list<GameObject*> GameObjectList;
         GameObjectList m_gameObj;
 
+        typedef std::list<AreaTrigger*> AreaTriggerList;
```

Comment:
Well Unit::_UnregisterAreaTrigger use the .remove function, which is lacking in std::vector